app/src/main/kotlin/com/softteco/template/MainViewModel.kt

```diff
@@ -9,31 +9,29 @@ import com.softteco.template.ui.feature.settings.PreferencesKeys
 import com.softteco.template.ui.theme.ThemeMode
 import com.softteco.template.utils.AppDispatchers
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.stateIn
-import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class MainViewModel @Inject constructor(
     dataStore: DataStore<Preferences>,
     appDispatchers: AppDispatchers,
-    private val profileRepository: ProfileRepository,
+    profileRepository: ProfileRepository,
 ) : ViewModel() {
 
     val theme = dataStore.data
         .map { it[PreferencesKeys.THEME_MODE] ?: ThemeMode.SystemDefault.value }
         .stateIn(viewModelScope, SharingStarted.Lazily, ThemeMode.SystemDefault.value)
 
-    val isUserLoggedIn = MutableStateFlow<Boolean?>(null)
-
-    init {
-        viewModelScope.launch(appDispatchers.io) {
-            profileRepository.getUser().let { result ->
-                isUserLoggedIn.value = result.isSuccess
-            }
-        }
-    }
+    val isUserLoggedIn: StateFlow<Boolean?> = profileRepository.getUser().map { result ->
+        result.isSuccess
```

Comment:
We do not receive an event here when we request a user from the Profile screen and the token is invalid or expired. Because of this, the Profile screen remains in the Loading state after the error is displayed

Reply:
As I understand it, the idea was that Graph would be changed to `Login` in case Flow `getUser()` does not return Profile in MianViewModel. The InvalidToken error is only handled in the ProfileViewModel and Graph is not changed in this case.

Reply:
Yes, you are right, but I suggest doing it in a separate ticket, as it seems out of the scope. The current implementation works as it worked before.

Reply:
But in the upstream implementation, the invalid token is handled correctly, and the app does not get stuck on the profile screen

Reply:
Good catch. Fixed and added ticket for further improvements  https://github.com/SoftTeco/AndroidAppTemplate/issues/187